homeassistant/components/vizio/config_flow.py

```diff
@@ -0,0 +1,131 @@
+            ): vol.All(str, vol.Lower, vol.In(["tv", "soundbar"])),
+            vol.Optional(
+                CONF_ACCESS_TOKEN, default=input_dict.get(CONF_ACCESS_TOKEN, "")
+            ): str,
+            vol.Optional(
+                CONF_VOLUME_STEP,
+                default=input_dict.get(CONF_VOLUME_STEP, DEFAULT_VOLUME_STEP),
+            ): vol.All(vol.Coerce(int), vol.Range(min=1, max=10)),
+        }
+    )
+
+
+class VizioConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a Vizio config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self) -> None:
+        """Initialize config flow."""
+        self.import_schema = None
+        self.user_schema = None
+
+    async def async_step_user(
+        self, user_input: Dict[str, Any] = None
+    ) -> Dict[str, Any]:
+        """Handle a flow initialized by the user."""
+        errors = {}
+
+        if user_input is not None:
+            await self.async_set_unique_id(unique_id=user_input[CONF_HOST])
+
+            # Store current values in case setup fails and user needs to edit
+            self.user_schema = update_schema_defaults(user_input)
+
+            # Check if new config entry matches any existing config entries
+            for entry in self.hass.config_entries.async_entries(DOMAIN):
+                if entry.data[CONF_HOST] == user_input[CONF_HOST]:
+                    errors[CONF_HOST] = "host_exists"
+                    break
+
+                if entry.data[CONF_NAME] == user_input[CONF_NAME]:
+                    errors[CONF_NAME] = "name_exists"
+                    break
+
+            if not errors:
+                try:
+                    # Ensure schema passes custom validation, otherwise catch exception and add error
+                    validate_auth(user_input)
+
+                    # Ensure config is valid for a device
+                    if not await VizioAsync.validate_ha_config(
+                        user_input[CONF_HOST],
+                        user_input.get(CONF_ACCESS_TOKEN),
+                        user_input[CONF_DEVICE_CLASS],
+                    ):
+                        errors["base"] = "cant_connect"
+                except vol.Invalid:
+                    errors["base"] = "tv_needs_token"
+
+            if not errors:
+                return self.async_create_entry(
+                    title=user_input[CONF_NAME], data=user_input
+                )
+
+        schema = self.user_schema or self.import_schema or update_schema_defaults({})
+
+        return self.async_show_form(step_id="user", data_schema=schema, errors=errors)
+
+    async def async_step_import(self, import_config: Dict[str, Any]) -> Dict[str, Any]:
+        """Import a config entry from configuration.yaml."""
+        # Abort flow if flow in progress or component with host as unique ID matches new config entry
+        try:
+            if await self.async_set_unique_id(
```

Comment:
Since the host is part of the saved entry data we can use that to check for same entry. We don't set a unique id at this point for import.